plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,454 @@
+    - Both O(dscp_mappings.dscp_from) and O(dscp_mappings.dscp_to) cannot be set to C(unspecified).
+    - Providing an empty list will remove the  O(cos_mappings=[]) from the Custom QoS Policy.
+    type: list
+    elements: dict
+    suboptions:
+      dscp_from:
+        description:
+        - The starting encoding point of the DSCP range.
+        - Defaults to C(unspecified) when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ from ]
+      dscp_to:
+        description:
+        - The ending encoding point of the DSCP range.
+        - Defaults to C(unspecified) when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target encoding point for egressing traffic.
+        - Defaults to C(unspecified) when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS traffic type for the egressing traffic.
+        - Defaults to C(unspecified) when unset during creation.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+      qos_priority:
+        description:
+        - The QoS priority level to which the DSCP values will be mapped.
+        - Defaults to C(unspecified) when unset during creation.
+        type: str
+        choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+        aliases: [ priority, prio ]
+  cos_mappings:
```

Comment:
How do you perform the update operation with the existing list of `dscp_mappings`? Will the old entries be replaced by the new entries?

Could you add a note to explain the update logic?

Reply:
same explanation as before